server/src/main/java/org/candlepin/async/LoggingJob.java

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright (c) 2009 - 2019 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.async;
+
+import org.candlepin.pinsetter.core.model.JobStatus;
+import org.slf4j.MDC;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Stores the job's unique ID and other information in logger's thread local MDC, which will
+ * automatically add it to all log entries executed for this job.
+ */
+public class LoggingJob implements AsyncJob {
+
+    private final AsyncJob job;
+
+    public LoggingJob(final AsyncJob job) {
+        this.job = requireNonNull(job);
+    }
+
+    @Override
+    public Object execute(final JobExecutionContext jdata) throws JobExecutionException {
+        MDC.put("requestType", "job");
+        if (jdata != null) {
+            final JobDataMap map = jdata.getJobData();
+
+            String requestUuid = null;
+            String orgKey = null;
+            String orgLogLevel = null;
+
+            if (map != null) {
+                requestUuid = map.getAsString("requestUuid");
```

Comment:
Do we have a constant for "requestUuid" somewhere?

Reply:
I was looking around since the other two values had a constant but haven't found any. I can create one. I am just not sure where to put it. Is there any common place for constants?